pkg/core/mempool/hashmap.go

```diff
@@ -235,6 +235,37 @@ func (m *HashMap) RangeSort(fn func(k txHash, t TxDesc) (bool, error)) error {
 	return nil
 }
 
+// ContainAnyNullifiers implements Pool.ContainAnyNullifiers.
+func (m *HashMap) ContainAnyNullifiers(nullifiers [][]byte) (bool, []byte) {
+	if len(nullifiers) == 0 {
+		return false, nil
+	}
+
+	m.lock.RLock()
+	defer m.lock.RUnlock()
+
+	for _, v := range m.data {
+		d, err := v.tx.Decode()
+		if err != nil {
+			continue
+		}
```

Comment:
should we really fail silently here? What to do if data is corrupted?

Reply:
`m.data` represents transactions that pass all checks and are accepted in mempool. If for some odd case `m.data` has accepted a corrupted tx, this (ContainAnyNullifiers) should not fail here, disallowing acceptance of other valid transactions.